### ISSUE TITLE
CKV2_GCP_4 False positive

### ISSUE THREAD

**Issue by brettcurtis** [bridgecrewio/checkov#2142]
**Describe the bug**

google_logging_project_sink fails on CKV2_GCP_4 if the bucket being used for logging sync has a variable set for retention_policy is_locked.

**To Reproduce**
You can at look this module: https://github.com/lzysh/terraform-google-project/blob/main/main.tf#L151 

This passes:
```hcl
  retention_policy {
    retention_period = 1800
    #is_locked        = var.bucket_lock
    is_locked        = true
  }
```

This fails:

```hcl
  retention_policy {
    retention_period = 1800
    is_locked        = var.bucket_lock
    #is_locked        = true
  }
```

**Expected behavior**
Both should pass since default variable is true.

**Additional context**
It also alerts on the follow resource:
![image](https://user-images.githubusercontent.com/1610100/147167477-4f5589c5-b738-4160-9627-7c2f7ae29290.png)

not the bucket resource, so that's a bit confusing.

**Comment by gruebel** [comment]
Hi @brettcurtis for the second question I can. give you an answer. Since you can decide what the destination for the sink is, your bucket is not the problem, only the sink, if it has a connection to bucket and of course it needs to be locked. Hope, that clarifies, why we mark the sink as not complaint with the check. 

It looks like you are using the VS Code extension, right? if you scan your code via `checkov` CLI, then it should work, at least for me 😄 If you can confirm this, then please open an issue here https://github.com/bridgecrewio/checkov-vscode

```shell
Check: CKV2_GCP_4: "Ensure that retention policies on log buckets are configured using Bucket Lock"
        PASSED for resource: google_logging_project_sink.cis_gcp_2_2_logging
        File: /main.tf:1-6
        Guide: https://docs.bridgecrew.io/docs/ensure-that-retention-policies-on-log-buckets-are-configured-using-bucket-lock

```

**Comment by brettcurtis** [comment]
Hmm, I tested with checkov (2.0.690) and do see it there:

![image](https://user-images.githubusercontent.com/1610100/147481756-72ebf253-da0e-4973-bb53-d7b2df27a182.png)

**Comment by gruebel** [comment]
ok, that was tricky 😄 the issue is actually right in your repo
https://github.com/lzysh/terraform-google-project/blob/29025c9f8ee8e019fd348b1f9636a0ddbe633310/test/fixtures/default_project/main.tf#L9
`checkov` parses the whole folder and all the sub folders and then tries to create relations between them. If you don't want it to include the subfolder just use the `--skip-path` flag and it will work 😉

**Comment by brettcurtis** [comment]
doh, oh yeah I forgot about that. Thats just my test code so I don't want to lock that bucket since I delete it right after the tests run.  So we are back to a vscode bug, and I opened https://github.com/bridgecrewio/checkov-vscode/issues/84
